pkg/services/authn/clients/proxy.go

```diff
@@ -103,8 +104,14 @@ func (c *Proxy) Authenticate(ctx context.Context, r *authn.Request) (*authn.Iden
 				// if we for some reason cannot find the user we proceed with the normal flow, authenticate with ProxyClient
 				// and perform syncs
 				if usr != nil {
-					c.log.FromContext(ctx).Debug("User was loaded from cache, skip syncs", "userId", usr.UserID)
-					return authn.IdentityFromSignedInUser(authn.NamespacedID(authn.NamespaceUser, usr.UserID), usr, authn.ClientParams{SyncPermissions: true}, login.AuthProxyAuthModule), nil
+					// Get the authentication information for the user
+					authedBy, err := c.authInfoService.GetAuthInfo(ctx, &login.GetAuthInfoQuery{UserId: usr.UserID})
+					if err != nil || authedBy == nil {
+						c.log.FromContext(ctx).Warn("Cached user had no valid auth info", "error", err, "userId", string(entry))
+					} else {
+						c.log.FromContext(ctx).Debug("User was loaded from cache, skip syncs", "userId", usr.UserID)
+						return authn.IdentityFromSignedInUser(authn.NamespacedID(authn.NamespaceUser, usr.UserID), usr, authn.ClientParams{SyncPermissions: true}, authedBy.AuthModule), nil
+					}
```

Comment:
Do we really need this part. We are not doing any syncing.

If we want to always display the correct authenticated by, even when using session, maybe this should be part of [user fetch hook](https://github.com/grafana/grafana/blob/main/pkg/services/authn/authnimpl/sync/user_sync.go#L109) instead. Otherwise only auth proxy would always display this info correctly

Then we should probably use that hook to fetch all user info instead of doing it manually here?

Reply:
:+1: let me see how that looks like